ipn/ipnlocal/local.go

```diff
@@ -1105,10 +1105,19 @@ func (b *LocalBackend) SetControlClientStatus(c controlclient.Client, st control
 		prefsChanged = true
 	}
 
+	// Until recently, we did not store the account's tailnet name. So check if this is the case,
+	// and backfill it on incoming status update.
+	if b.pm.missingCurrentProfileTailnetName() && st.NetMap.TailnetName() != "" {
```

Comment:
I'd have this check for any of the network profile fields being empty

Reply:
Incase we add more in the future, we get the filling for free

Reply:
Any suggestions as to how? The only way I can think of is by using reflect which might be overkill. Also, some future fields might be optional, so their emptiness does not necessarily indicate them not being backfilled, reflect could use some struct tag to indicate it but that also might be an overkill 😅

Reply:
Maybe add an IsZero func to network profile - that would need to be updated when adding new fields, but if it lives next to the struct definition, it's a little harder to miss

Reply:
Will do 👍🏼 might bikeshed the name a little 🙏🏼  because in the future it may not be "zero" but could still require a backfill. So calling it `RequiresBackfill() bool` for now, but happy to take other suggestions

Reply:
Works for me! But yeah I was thinking probably just directly comparing each of the fields for emptiness would probably be okay for now.